extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -164,112 +187,121 @@ private void restartJobs() {
         if (schedulerFuture != null && !schedulerFuture.isCancelled()) {
             schedulerFuture.cancel(true);
         }
-
-        schedulerFuture = scheduler.schedule(new Runnable() {
-            @Override
-            public void run() {
+        if (localExecutor != null) {
+            localExecutor.shutdownNow();
+        }
+        localExecutor = Executors.newSingleThreadExecutor();
```

Comment:
This executor instance has been introduced to execute the immediate jobs (daily job and positional job). The lifecycles of handling these jobs are different than the `scheduler` in handler.